src/sdam/topology.ts

```diff
@@ -626,7 +628,7 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
 
   /** Start a logical session */
   startSession(options: ClientSessionOptions, clientOptions?: MongoOptions): ClientSession {
-    const session = new ClientSession(this, this.s.sessionPool, options, clientOptions);
+    const session = new ClientSession(this.client, this.s.sessionPool, options, clientOptions);
```

Comment:
It appears this method is the only reason we need to store a reference to the client on the topology class.  Additionally, the client has a reference to the topology already.

How would you feel about removing this method and moving the logic into `client.startSession`?  That also prevents the circular client -> topology -> client reference.  

This method isn't marked as internal but Topology is so I think we're okay to make that change?

Reply:
I attempted this, it's quite complex but we could do it in a follow up. I'll update when I've made a ticket

Reply:
from slack - startSession on the client starts an explicit session every time whereas this can start both an explicit session or an implicit session.  So any internal consumers of Topology.startSession would have to work around the client starting the explicit session.